### PR TITLE
OPM - fix issue with ordering

### PR DESCRIPTION
**Pull request by lwyszomi**
@czue @esoergel 
http://manage.dimagi.com/default.asp?175650#972812

### CODE REVIEW

File: custom/opm/reports.py [hunk 1]
```diff
@@ -738,7 +738,7 @@ class CaseReportMixin(object):
 
     @memoized
     def column_index(self, key):
-        for i, (k, _, _) in enumerate(self.model.method_map):
+        for i, (k, _, _, _) in enumerate(self.model.method_map):
```
Comment (dimagimon) [inline custom/opm/reports.py]: F402 import '_' from line 21 shadowed by loop variable

File: custom/opm/reports.py [hunk 2]
```diff
@@ -738,7 +738,7 @@ class CaseReportMixin(object):
 
     @memoized
     def column_index(self, key):
-        for i, (k, _, _) in enumerate(self.model.method_map):
+        for i, (k, _, __, ___) in enumerate(self.model.method_map):
```
Comment (dimagimon) [inline custom/opm/reports.py]: F402 import '_' from line 21 shadowed by loop variable

**Comment by czue** [comment]
I have it in my head that you also have to transform the values in the rows for this to work, but didn't see that change. assuming you tested and that's not necessary?

**Comment by lwyszomi** [comment]
@czue Yes I tested this and all works great.

**Comment by lwyszomi** [comment]
@czue please don't merge, I found bug

File: custom/opm/utils.py [hunk 3]
```diff
@@ -84,3 +84,13 @@ def get_matching_users(awcs=None, gps=None, blocks=None):
...
+
+
+def numeric_fn(val):
+    try:
+        sort_val = int(val)
+    except ValueError:
+        sort_val = -1
+    except TypeError:
+        sort_val = -1
+    return {'sort_key': sort_val, 'html': val}
```
Comment (dimagimon) [inline custom/opm/utils.py]: W292 no newline at end of file

**Comment by esoergel** [comment]
@lwyszomi There's a merge conflict, could you resolve please?

**Comment by esoergel** [comment]
Also, we have a label you can apply to PRs when you don't want them merged yet:  `[Open for review: hold off on merge]`.  Do you have the ability to add labels to the PR if you're not a repository owner?

**Comment by lwyszomi** [comment]
@czue @esoergel Updated. I don't have access to add lables to PR.

File: custom/opm/utils.py [hunk 4]
```diff
@@ -84,3 +84,13 @@ def get_matching_users(awcs=None, gps=None, blocks=None):
         user for user in UserSqlData().get_data()
         if user[key] in selected
     ]
+
+
+def numeric_fn(val):
```
Comment (czue) [inline custom/opm/utils.py]: (this is what i was thinking of)

Comment (lwyszomi) [inline custom/opm/utils.py]: Yes, you had right with this.

**Comment by lwyszomi** [comment]
@czue @esoergel Ready to merge

File: custom/opm/incentive.py [hunk 5]
```diff
@@ -41,29 +44,34 @@ def __init__(self, worker, report, case_data=None, form_data=None):
...
                 dates = opm_case.data_provider.get_dates_in_range(opm_case.owner_id,
                                                                   opm_case.reporting_window_start,
                                                                   opm_case.reporting_window_end)
 
                 self.service_forms_count = 'yes' if dates else 'no'
-            self.growth_monitoring_count = len(filter(lambda row: row.growth_calculated_aww, case_data))
+            monitoring_count = len(filter(lambda row: row.growth_calculated_aww, case_data))
         else:
             self.women_registered = None
             self.children_registered = None
```
Comment (czue) [inline custom/opm/incentive.py]: out of curiosity any reason you're using `None` for these but `0` for `growth_monitoring_count`?

**Comment by czue** [comment]
wow, this was quite a bit more complicated than I expected! thanks

**Pull request merged**